From 6db0a5d3c237fe39be06bc09fc4532b5d3312797 Mon Sep 17 00:00:00 2001
From: Volker Lendecke <vl@samba.org>
Date: Tue, 19 Aug 2014 14:32:15 +0000
Subject: [PATCH 292/321] smbd: Properly initialize mangle_hash

[Bug 10782] mangle_hash() can fail to initialize charset (smbd crash).

https://bugzilla.samba.org/show_bug.cgi?id=10782

Signed-off-by: Volker Lendecke <vl@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>

Autobuild-User(master): Jeremy Allison <jra@samba.org>
Autobuild-Date(master): Tue Aug 26 01:30:38 CEST 2014 on sn-devel-104

(cherry picked from commit e914c2c52db7ecf3bb2a3860820c5cfe8812696e)
---
 source3/smbd/mangle_hash.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source3/smbd/mangle_hash.c b/source3/smbd/mangle_hash.c
index bafcd03d015..35a098f5049 100644
--- a/source3/smbd/mangle_hash.c
+++ b/source3/smbd/mangle_hash.c
@@ -706,6 +706,10 @@ const struct mangle_fns *mangle_hash_init(void)
 {
 	mangle_reset();
 
+	if (chartest == NULL) {
+		init_chartest();
+	}
+
 	/* Create the in-memory tdb using our custom hash function. */
 	tdb_mangled_cache = tdb_open_ex("mangled_cache", 1031, TDB_INTERNAL,
 				(O_RDWR|O_CREAT), 0644, NULL, fast_string_hash);
-- 
2.14.3

